mne/viz/eyetracking/heatmap.py

```diff
@@ -0,0 +1,167 @@
+        The width dimension of the plot canvas. For example, if the eyegaze data units
+        are pixels, and the participant screen resolution was 1920x1080, then the width
+        should be 1920.
+    height : int
+        The height dimension of the plot canvas. For example, if the eyegaze data units
+        are pixels, and the participant screen resolution was 1920x1080, then the height
+        should be 1080.
+    sigma : int | float
+        The amount of smoothing applied to the heatmap data. If ``None``,
+        no smoothing is applied. Default is 1.
+    cmap : matplotlib colormap | str | None
+        The :class:`~matplotlib.colors.Colormap` to use. Defaults to ``None``, meaning
+        the colormap will default to matplotlib's default.
+    alpha : int | float | None
+        The transparency value of the heatmap. If ``None``, the alpha value is set to 1,
+        meaning the heatmap colors are fully opaque. Default is ``None``.
+    vmin : float | None
+        The minimum value for the colormap. The unit is seconds, for dwell time
+        to the bin coordinate. If ``None``, the minimum value is set to the
+        minimum value of the heatmap. Default is ``None``.
+    vmax : float | None
+        The maximum value for the colormap. The unit is seconds, for dwell time
+        to the bin coordinate. If ``None``, the maximum value is set to the
+        maximum value of the heatmap. Default is ``None``.
+    axes : matplotlib.axes.Axes | None
+        The axes to plot on. If ``None``, a new figure and axes are created.
+        Default is ``None``.
+    show : bool
+        Whether to show the plot. Default is ``True``.
+
+    Returns
+    -------
+    fig : instance of matplotlib.figure.Figure
+        The resulting figure object for the heatmap plot.
+
+    Notes
+    -----
+    .. versionadded:: 1.6
+    """
+    from mne import BaseEpochs
+    from mne._fiff.pick import _picks_to_idx
+
+    _validate_type(epochs, BaseEpochs, "epochs")
+
+    pos_picks = _picks_to_idx(epochs.info, "eyegaze")
+    gaze_data = epochs.get_data(picks=pos_picks)
+    gaze_ch_loc = np.array([epochs.info["chs"][idx]["loc"] for idx in pos_picks])
+    x_data = gaze_data[:, np.where(gaze_ch_loc[:, 4] == -1)[0], :]
+    y_data = gaze_data[:, np.where(gaze_ch_loc[:, 4] == 1)[0], :]
+
+    if x_data.shape[1] > 1:  # binocular recording. Average across eyes
+        logger.info("Detected binocular recording. Averaging positions across eyes.")
+        x_data = np.nanmean(x_data, axis=1)  # shape (n_epochs, n_samples)
+        y_data = np.nanmean(y_data, axis=1)
+    x_data, y_data = x_data.flatten(), y_data.flatten()
+    canvas = np.vstack((x_data, y_data)).T  # shape (n_samples, 2)
+
+    # Create 2D histogram
+    hist, _, _ = np.histogram2d(
+        canvas[:, 0],
+        canvas[:, 1],
+        bins=(width, height),
+        range=[[0, width], [0, height]],
+    )
+    hist = hist.T  # transpose to match screen coordinates. i.e. width > height
```

Comment:
I think this is equivalent, no?
```suggestion
    # Bin into image-like format
    hist, _, _ = np.histogram2d(
        canvas[:, 1],
        canvas[:, 0],
        bins=(height, width),
        range=[[0, height], [0, width]],
    )
```

Reply:
Let's find out!